src/lang/nl.rs

```diff
@@ -0,0 +1,727 @@
+use crate::{num2words::Num2Err, Currency, Language};
+use num_bigfloat::BigFloat;
+
+pub struct Dutch {}
+
+const UNITS: [&str; 9] = [
+    "één", "twee", "drie", "vier", "vijf", "zes", "zeven", "acht", "negen",
+];
+
+const TENS: [&str; 9] = [
+    "tien", "twintig", "dertig", "veertig", "vijftig", "zestig", "zeventig", "tachtig", "negentig",
+];
+
+const TEENS: [&str; 10] = [
+    "tien",
+    "elf",
+    "twaalf",
+    "dertien",
+    "veertien",
+    "vijftien",
+    "zestien",
+    "zeventien",
+    "achtien",
+    "negentien",
+];
+
+// As defined by the AHD4, CED, RHD2, W3 and UM authorities
+// For more information, see
+// https://en.wikipedia.org/wiki/Names_of_large_numbers
+const MEGAS: [&str; 21] = [
+    "duizend",
+    "miljoen",
+    "miljard",
+    "biljoen",
+    "biljard",
+    "triljoen",
+    "triljard",
+    "quadriljoen",
+    "quadriljard",
+    "quintiljoen",
+    "quintiljard",
+    "sextiljoen",
+    "sextiljard",
+    "septiljoen",
+    "septiljard",
+    "octiljoen",
+    "octiljard",
+    "noniljoen",
+    "noniljard",
+    "deciljoen",
+    "deciljard",
+];
+
+fn space_words(words: &mut Vec<String>) {
+    for x in (0..words.len()).rev() {
+        let word: String = words.get(x).unwrap().clone();
+        if ["minus"].contains(&word.as_str()) {
+            words.insert(x + 1, " ".to_string());
+        } else if ["komma"].contains(&word.as_str()) {
+            words.insert(x, " ".to_string());
+        } else if MEGAS.contains(&word.as_str()) {
+            if x != words.len() - 1 {
+                words.insert(x + 1, " ".to_string());
+            }
+            if !word.eq("duizend") {
+                words.insert(x, " ".to_string());
+            }
+        }
+    }
+}
+
+impl Dutch {
+    pub fn new() -> Self {
+        Self {}
+    }
+
+    fn currencies(&self, currency: Currency) -> String {
+        currency.default_string(false)
+    }
+
+    fn cents(&self, currency: Currency) -> String {
+        currency.default_subunit_string("cent{}", false)
+    }
+
+    fn split_thousands(&self, mut num: BigFloat) -> Vec<u64> {
+        let mut thousands = Vec::new();
+        let bf_1000 = BigFloat::from(1000);
+
+        while !num.is_zero() {
```

Comment:
Addressed in the PR: https://github.com/Ballasi/num2words/pull/33 , It is at least advisable truncate into an integer on each call to avoid issues discussed at Issue https://github.com/Ballasi/num2words/issues/11